projects/packages/search/src/dashboard/hooks/use-product-checkout-workflow.jsx

```diff
@@ -0,0 +1,90 @@
+import restApi from '@automattic/jetpack-api';
+import { getProductCheckoutUrl } from '@automattic/jetpack-components';
+import { useConnection, CONNECTION_STORE_ID } from '@automattic/jetpack-connection';
+import { useDispatch } from '@wordpress/data';
+import { useEffect, useState } from 'react';
+
+const {
+	registrationNonce,
+	apiRoot,
+	apiNonce,
+	siteSuffix: defaultSiteSuffix,
+} = window?.JP_CONNECTION_INITIAL_STATE ? window.JP_CONNECTION_INITIAL_STATE : {};
+
+/**
+ * Custom hook that performs the needed steps
+ * to concrete the checkout workflow.
+ *
+ * @param {object} props              - The props passed to the hook.
+ * @param {string} props.productSlug  - The WordPress product slug.
+ * @param {string} props.redirectUrl  - The URI to redirect to after checkout.
+ * @param {string} [props.siteSuffix] - The site suffix.
+ * @param {Function} props.siteProductAvailabilityHandler - The function used to check whether the site already has the requested product. This will be checked after registration and the checkout page will be skipped if the promise returned resloves true.
+ * @param {Function} props.from       - The plugin slug initiated the flow.
+ * @param {Function} props.isWpcom    - Whether it's WPCOM site.
+ * @returns {Function}				  - The useEffect hook.
+ */
+export default function useProductCheckoutWorkflow( {
```

Comment:
Do we want to port this back to the `@automattic/jetpack-connection` package in the future?

Reply:
Possibly, I didn't change the original code in the connections package just to avoid the risk to affect other products.